src/main/java/org/opentripplanner/model/TimetableSnapshot.java

```diff
@@ -262,6 +372,12 @@ public TimetableSnapshot commit(TransitLayerUpdater transitLayerUpdater, boolean
     TimetableSnapshot ret = new TimetableSnapshot(
       Map.copyOf(timetables),
       Map.copyOf(realtimeAddedTripPattern),
+      Map.copyOf(realtimeAddedRoutes),
+      Map.copyOf(realTimeAddedTrips),
+      Map.copyOf(realTimeAddedPatternForTrip),
+      ImmutableSetMultimap.copyOf(realTimeAddedPatternsForRoute),
+      Map.copyOf(realTimeAddedTripOnServiceDateById),
+      Map.copyOf(realTimeAddedTripOnServiceDateForTripAndDay),
       ImmutableSetMultimap.copyOf(patternsForStop),
```

Comment:
I believe this is here because `TimetableSnapshot` is sometimes mutable (when it acts as a buffer) and sometimes immutable (when it's published and used by routing threads).

Reply:
Leonard is right, this constructor is used both for the read/write (i.e. buffer) and read-only (i.e. snapshot) `TimetableSnapshot`

Reply:
A comment on top of this code explaining that would be great. But, if we plan to fix it - we can live without that for now.